### PR TITLE
[CDAP-19529] Add dependency class localization for running programs on kubernetes

### PR DESCRIPTION
**Pull request by dli357**
This allows us to localize dependency classes and jars via the FileLocalizer which is necessary for classes which are required to run on cdap-kubernetes.

### CODE REVIEW

File: cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/DistributedSparkProgramRunner.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ protected void setupLaunchConfig(ProgramLaunchConfig launchConfig, Program progr
 
     launchConfig
       .addExtraEnv(extraEnv)
-      .addExtraDependencies(SparkProgramRuntimeProvider.class)
+      .addExtraDependencies(SparkProgramRuntimeProvider.class, SparkTwillRunnable.class)
```
Comment (albertshau) [inline cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/DistributedSparkProgramRunner.java:177]: Is this change needed? If it is, I'm a little confused how the existing Hadoop code path works without SparkTwillRunnable.class as a dependency.

**Comment by dli357** [comment]
Closing this PR in favor of #14536.
